cpp/src/datetime/datetime_ops.cu

```diff
@@ -68,98 +67,158 @@ struct extract_component_operator {
   }
 };
 
-template <datetime_component Component>
-struct launch_extract_component {
+// Round up the date to the last day of the month and return the
+// date only (without the time component)
+struct extract_last_day_of_month {
+  CUDA_DEVICE_CALLABLE auto days_in_month(simt::std::chrono::month mon,
+                                          bool is_leap_year) const -> uint8_t {
+    using namespace simt::std::chrono;
+    switch (static_cast<unsigned>(mon)) {
+      case January   : return 31;
+      case February  : return is_leap_year ? 29 : 28;
+      case March     : return 31;
+      case April     : return 30;
+      case May       : return 31;
+      case June      : return 30;
+      case July      : return 31;
+      case August    : return 31;
+      case September : return 30;
+      case October   : return 31;
+      case November  : return 30;
+      case December  : return 31;
+      default        : return 0;
```

Comment:
Clang shows errors in this switch statement:

![Screenshot_2020-03-28_13-59-22](https://user-images.githubusercontent.com/178183/77833733-cce74700-70fc-11ea-9164-1f8f613ffde7.png)

Proposing this fix to quiet the noise:
```suggestion
      case static_cast<unsigned>(January)   : return 31;
      case static_cast<unsigned>(February)  : return is_leap_year ? 29 : 28;
      case static_cast<unsigned>(March)     : return 31;
      case static_cast<unsigned>(April)     : return 30;
      case static_cast<unsigned>(May)       : return 31;
      case static_cast<unsigned>(June)      : return 30;
      case static_cast<unsigned>(July)      : return 31;
      case static_cast<unsigned>(August)    : return 31;
      case static_cast<unsigned>(September) : return 30;
      case static_cast<unsigned>(October)   : return 31;
      case static_cast<unsigned>(November)  : return 30;
      case static_cast<unsigned>(December)  : return 31;
      default                               : return 0;
```

Reply:
@trxcllnt - please try the new version.

Reply:
Yep looks good @sriramch thanks!